### PR TITLE
[WIP] Switch to docute documentation site generator

### LINKED ISSUE #1974: Docs: switch from readthedocs to docute
I've spent a while investigating and assessing various options to replace readthedocs, including but not limited to docusaurus, docsify, docpress, gitbook and vuepress.

I really like the docsify approach and then discovered the similar but-with-more-bells-attached [docute](https://github.com/egoist/docute) and it appears to support pretty much everything we need (and more):

* retrieves *versioned* markdown files from GitHub, optionally via jsdelivr
* customisable URL router to redirect existing readthedocs URLs
* easy to define navigation/sidebar/TOC
* custom components/banner/footer via Vue
* sends anonymous usage data to GA

(For examples of problems reported about readthedocs, please see #1451 and #1971.)

### PR DESCRIPTION
**Pull request by sebinbenjamin**
Sets up docute to replace readthedocs. Fixes #1974 

@lovell  I've tried to include everything from https://sharp.pixelplumbing.com . Please have a look.

Peek 2019-11-18 16-20

### Tasks
- [x] Retrieves versioned markdown files from GitHub, via jsdelivr
- [x] Customise URL router to redirect to existing readthedocs URLs
- [x] Define navigation bar (with logo) and sidebar
- [x] Custom Footer
- [ ] Sends anonymous usage data to [GA](https://github.com/egoist/docute-plugins/tree/master/packages/google-analytics) .
- [ ] Verify correctness of links in all pages. 

I think the following might also be good. I could create a seperate issue/PR if they will be useful.
### Additional
- [ ] Search feature - Maybe https://community.algolia.com/docsearch/ . Related https://github.com/egoist/docute/issues/21, https://github.com/egoist/docute/pull/261
- [ ] Configure documentation.js - Is it possible to add #title to generated .md files, like in `install.md` and `performance.md`
- [ ] Offline support - https://docute.org/guide/offline-support.

### CODE REVIEW

**Comment by sebinbenjamin** [comment]
Note: Links are currently pointed to https://github.com/sebinbenjamin/sharp/edit/master/docs as I needed a new tag v0.23.4-alpha for testing. Needs to be changed before closing PR.

Also do share the GA Tracking ID.

**Comment by lovell** [comment]
This is amazing, thank you.

For the previous versions, readthedocs will be going away, so ideally these will come from GitHub too. The markdown files in docs/ are all tagged by version so I was hoping we could use something like https://docute.org/guide/customization#versioning for this.

The GA code is UA-13034748-12 (this is the user_analytics_code from https://sharp.pixelplumbing.com/en/stable/readthedocs-data.js)

Sadly documentationjs does not produce page titles for markdown output but we could define a custom plugin to replace the `<!-- Generated by ...` with `<h1 class="page-title">Name of API section</h1>` a bit like the example in https://docute.org/guide/plugin

File: docs/image/sharp-logo.svg [hunk 1]
```diff
@@ -1,5 +1 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="86 86 550 550">
-<!-- Copyright 2019 Lovell Fuller. This work is licensed under the Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0) License. -->
-<path fill="none" stroke="#9c0" stroke-width="80" d="M258.411 285.777l200.176-26.8M244.113 466.413L451.44 438.66M451.441 438.66V238.484M451.441 88.363v171.572l178.725-23.917M270.323 255.602V477.22M272.71 634.17V462.591L93.984 486.515"/>
-<path fill="none" stroke="#090" stroke-width="80" d="M451.441 610.246V438.66l178.725-23.91M269.688 112.59v171.58L90.964 308.093"/>
-</svg>
\ No newline at end of file
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="86 86 50 49.636" width="50" height="49.636" fill="none" stroke-width="7.275"><path d="M101.7 103.953l18.204-2.437M100.4 120.38l18.854-2.524m0 0V99.652m0-13.652v15.603l16.253-2.175m-32.724 1.78v20.154m.217 14.273v-15.603l-16.253 2.176" stroke="#9c0"/><path d="M119.264 133.46v-15.604l16.253-2.174m-32.782-27.478v15.603l-16.253 2.176" stroke="#090"/></svg>
```
Comment (lovell) [inline docs/image/sharp-logo.svg]: Please can the copyright and licensing notice remain in this file.

Comment (sebinbenjamin) [reply docs/image/sharp-logo.svg]: Oh ! Missed it. Sorry about that.

**Comment by sebinbenjamin** [comment]
> For the previous versions, readthedocs will be going away, so ideally these will come from GitHub too. The markdown files in docs/ are all tagged by version so I was hoping we could use something like https://docute.org/guide/customization#versioning for this.

I tried versioning but got stuck with the file name requirement. I think that docute by default [wants](https://docute.org/#how-does-it-work) the landing page markdown file to be named as `README.md`, but we have it named as `index.md` for `readthedocs`. I am not very familiar with vue.js components/plugins, but am guessing from this [code](https://github.com/egoist/docute/blob/c4d2bf9881d6a8fecf275395c0c0e5e88eeb49ca/src/utils/index.js#L35) that we would not be able to override it. I could try putting up a feature request at docute for a custom default landing page filename.  🤔

> The GA code is UA-13034748-12 (this is the user_analytics_code from https://sharp.pixelplumbing.com/en/stable/readthedocs-data.js)

Thanks. I'll just put the this in.

> Sadly documentationjs does not produce page titles for markdown output but we could define a custom plugin to replace the `<!-- Generated by ...` with `<h1 class="page-title">Name of API section</h1>` a bit like the example in https://docute.org/guide/plugin

Nice. Let me check it.

**Comment by lovell** [comment]
I've merged this to the `wit` WIP branch via commit https://github.com/lovell/sharp/commit/39256894354baab04cd573c6913162f46d371751

Rather than provide docs for loads of older versions, I've added `@since` tags to some of the API to indicate the version from which it was made available.

These new docs, for the forthcoming v0.24.0 release, can be temporarily viewed at https://pixelplumbing-sharp.firebaseapp.com/ for now. I've configured Firebase to handle all the redirects - see https://github.com/lovell/sharp/blob/wit/docs/firebase.json#L22

I'll update the main documentation URL to point to this after sharp v0.24.0 is released.

Thanks again for all your help with this @sebinbenjamin - I think this PR makes you the 100th contributor to sharp!

**Pull request closed without merging**